browser/brave_ads/request_ads_enabled_api_browsertest.cc

```diff
@@ -0,0 +1,293 @@
+      ui_test_utils::BROWSER_TEST_WAIT_FOR_LOAD_STOP);
+
+  bool enabled;
+  EXPECT_TRUE(ExecuteScriptAndExtractBool(
+      contents, kResolveRequestAdsEnabledPromise, &enabled));
+  EXPECT_FALSE(enabled);
+}
+
+IN_PROC_BROWSER_TEST_F(RequestAdsEnabledApiTestEnabled,
+                       ApiForIncognitoBrowser) {
+  GURL url = https_server()->GetURL(kAllowedDomain, "/simple.html");
+
+  Browser* incognito_browser = OpenURLOffTheRecord(browser()->profile(), url);
+
+  content::WebContents* contents =
+      incognito_browser->tab_strip_model()->GetActiveWebContents();
+  WaitForLoadStop(contents);
+  EXPECT_EQ(url, contents->GetURL());
+
+  bool has_api;
+  EXPECT_TRUE(ExecuteScriptAndExtractBool(
+      contents, kBraveRequestAdsEnabledExists, &has_api));
+  EXPECT_TRUE(has_api);
+
+  EXPECT_TRUE(ExecuteScript(contents, kBraveRequestAdsEnabled));
+
+  bool enabled;
+  EXPECT_TRUE(ExecuteScriptAndExtractBool(
+      contents, kResolveRequestAdsEnabledPromise, &enabled));
+  EXPECT_FALSE(enabled);
+}
+
+IN_PROC_BROWSER_TEST_F(RequestAdsEnabledApiTestEnabled,
+                       RunApiForWithoutUserGesture) {
+  GURL url = https_server()->GetURL(kAllowedDomain, "/simple.html");
+
+  ui_test_utils::NavigateToURL(browser(), url);
+  content::WebContents* contents =
+      browser()->tab_strip_model()->GetActiveWebContents();
+  WaitForLoadStop(contents);
+  EXPECT_EQ(url, contents->GetURL());
+
+  bool has_api;
+  EXPECT_TRUE(ExecuteScriptWithoutUserGestureAndExtractBool(
+      contents, kBraveRequestAdsEnabledExists, &has_api));
+  EXPECT_TRUE(has_api);
+
+  std::string status;
+  EXPECT_TRUE(ExecuteScriptWithoutUserGestureAndExtractString(
+      contents, kGetRequestAdsEnabledPromiseRejectReason, &status));
+  EXPECT_EQ(status, kUserGestureRejectReason);
+}
+
+IN_PROC_BROWSER_TEST_F(RequestAdsEnabledApiTestEnabled,
+                       ApiNotAvailableForUnknownHost) {
+  GURL url = https_server()->GetURL(kNotAllowedDomain, "/simple.html");
+
+  ui_test_utils::NavigateToURL(browser(), url);
+  content::WebContents* contents =
+      browser()->tab_strip_model()->GetActiveWebContents();
+  WaitForLoadStop(contents);
+  EXPECT_EQ(url, contents->GetURL());
+
+  bool has_api;
+  EXPECT_TRUE(ExecuteScriptAndExtractBool(
+      contents, kBraveRequestAdsEnabledExists, &has_api));
+  EXPECT_FALSE(has_api);
+}
+
+class RequestAdsEnabledApiTestDisabled : public RequestAdsEnabledApiTestBase {
+ public:
+  RequestAdsEnabledApiTestDisabled() {
+    feature_list_.InitAndDisableFeature(
+        brave_ads::features::kRequestAdsEnabledApi);
+  }
+};
+
+IN_PROC_BROWSER_TEST_F(RequestAdsEnabledApiTestDisabled,
+                       ApiNotAvailableWhenFeatureOff) {
+  GURL url = https_server()->GetURL(kAllowedDomain, "/simple.html");
+
+  ui_test_utils::NavigateToURL(browser(), url);
+  content::WebContents* contents =
+      browser()->tab_strip_model()->GetActiveWebContents();
+  WaitForLoadStop(contents);
```

Comment:
no need for this (this copypasted pattern is repeated through the whole codebase). `NavigateToURL` is blocking on its own

Reply:
Fixed